vassal-app/src/main/java/VASSAL/build/module/BasicLogger.java

```diff
@@ -319,7 +319,7 @@ public void queryNewLogFile(boolean atStart) {
 
     final GameModule g = GameModule.getGameModule();
 
-    if ((Boolean) g.getPrefs().getValue(prefName)) {
+    if ((boolean) g.getPrefs().getValue(prefName)) {
```

Comment:
Wait, not this one! This is one of those cases where you _do_ need `Boolean`, because the `getValue()` returns an ` Object`, so if it returns `null` you'll have `ClassCastException` due to the `boolean` cast.

Reply:
If I marked this one as wrong, that was a mistake.

Reply:
Oh hey, looking at this again---that's still wrong (but was that way before, so not your fault). When you have something that returns an `Object` which could be a `Boolean`, you need to check if it's equal to `Boolean.TRUE`, like so:
```
if (Boolean.TRUE.equals(g.getPrefs().getValue(prefName))) {
```
What can happen with
```
if ((Boolean) g.getPrefs().getValue(prefName)) {
```
is that you get a null and then a `NullPointerException` on the attempt to unbox the `Boolean`.